packages/synapse-react-client/src/components/AclEditor/AclEditor.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { Box, Collapse, Typography } from '@mui/material'
+import { ResourceAccessItem } from './ResourceAccessItem'
+import UserSearchBoxV2 from '../UserSearchBox/UserSearchBoxV2'
+import { ResourceAccess } from '@sage-bionetworks/synapse-types'
+import useUpdateAcl from './useUpdateAcl'
+import { TransitionGroup } from 'react-transition-group'
+import { PermissionLevel } from '../../utils/PermissionLevelToAccessType'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+
+export type AclEditorProps = {
+  isLoading: boolean
+  resourceAccessList: ResourceAccess[]
+  availablePermissionLevels: PermissionLevel[]
+  emptyText: React.ReactNode
+  addResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['addResourceAccessItem']
+  updateResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['updateResourceAccessItem']
+  removeResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['removeResourceAccessItem']
+}
+
+export function AclEditor(props: AclEditorProps) {
+  const {
+    isLoading,
+    resourceAccessList,
+    availablePermissionLevels,
+    emptyText,
+    addResourceAccessItem,
+    updateResourceAccessItem,
+    removeResourceAccessItem,
+  } = props
+  if (isLoading) {
+    return (
+      <Box display={'flex'} justifyContent={'center'} height={'150px'}>
+        <SynapseSpinner size={50} />
+      </Box>
+    )
+  }
+
+  return (
+    <Box>
+      <Box mb="30px">
+        <Typography variant="headline3" mb="10px">
+          Users and Teams with Permissions
+        </Typography>
+
+        {resourceAccessList.length === 0 ? (
+          <Typography variant="body1Italic">{emptyText}</Typography>
+        ) : (
+          <TransitionGroup>
+            {resourceAccessList.map(resourceAccess => {
+              return (
+                <Collapse key={resourceAccess.principalId}>
```

Comment:
Add `TransitionGroup`/`Collapse` to animate addition/removal of items. This feels much more natural in cases like this when the UI to add a row is disconnected from the rest of the form.

[Screencast from 07-11-2024 05:37:47 PM.webm](https://github.com/user-attachments/assets/56c51112-84e3-4cb6-9eb6-7cb5958a2cca)